### PR TITLE
[#2236] feat(core): Add the support of Group entity

### PR DESCRIPTION
**Pull request by qqqttt123**
### What changes were proposed in this pull request?

Add the support of Group entity

### Why are the changes needed?

Fix: #2236 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Add a new UT.

### CODE REVIEW

File: api/src/main/java/com/datastrato/gravitino/authorization/Group.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.List;
+
+/** The interface of a Group. The Group is the entity which contains users. */
+@Evolving
+public interface Group extends Auditable {
```
Comment (jerryshao) [inline api/src/main/java/com/datastrato/gravitino/authorization/Group.java:13]: How does Gravitino know which user belongs to which group?

Comment (qqqttt123) [reply api/src/main/java/com/datastrato/gravitino/authorization/Group.java:13]: We can get the groups by using user name in the Authenticator. We use user name to retrieve the user from the external user system like LDAP, RANGER, IAM.

I will add `GroupMappingServiceProvider` for accomplish this feature.
```
interface GroupMappingServiceProvider {
   List<String> getGroups(String user);
}
```

File: core/src/main/java/com/datastrato/gravitino/authorization/GroupManager.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import java.util.Collections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * GroupManager is used for add, remove and get groups from one metalake. GroupManager doesn't
+ * manage group, just sets up the relationship between the metalake and the group. Metalake is like
+ * a concept of the organization. `AddGroup` means that a group enter an organization.
+ */
+public class GroupManager {
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/authorization/GroupManager.java:31]: I think we can merge the User and GroupManager together as "UserGroupManager", WDYT?

Comment (qqqttt123) [reply core/src/main/java/com/datastrato/gravitino/authorization/GroupManager.java]: Done.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**